templates/service.yaml

```diff
@@ -1,11 +1,17 @@
 apiVersion: v1
 kind: Service
 metadata:
-  name: {{ .Release.Name }}-helper-service
+  name: {{ .Release.Name }}-lb
+  annotations:
+    service.beta.kubernetes.io/aws-load-balancer-type: nlb
+    service.beta.kubernetes.io/aws-load-balancer-scheme: internet-facing
+    service.beta.kubernetes.io/aws-load-balancer-nlb-target-type: instance
 spec:
-  type: NodePort
+  type: LoadBalancer
```

Comment:
just to clarify, is this a LoadBalancer per helper? thinking a little bit ahead, but I know we have concerns about shards connecting through a load balancer being a choke point (and also, it's not really doing any balancing).

Reply:
K8s requires some sort of Service to expose the Pods to the internet. From what I investigated, using this kind of LoadBalancer (NLB) suits our needs best.

This is an AWS Elastic Load Balancer, so I don't imagine this being a choke point. Also, this is a "simple" Network Load Balancer. This one only routes on the network layer and can't for instance do SSL termination (that's on a higher level).

Reply:
got it. this is fine for now, but having a single load balancer per helper may become an issue with multiple shards. @akoshelev has brought this up in the past

Reply:
is it really needed? adding extra hops even at L4 does not help with scale. I am a bit surprised that we must have a proxy in front of our hosts but I guess that is K8-specific quirk?

Reply:
This is one of the different options K8s provides. Here are more details: https://kubernetes.io/docs/concepts/services-networking/

I'm of the opinion that having some kind of Load Balancer will be necessary to achieve High Availability and be able to hot switch between hosts if there's a new deployment.

The NodePort approach from before had this "weird" behavior, where a Pod (a helper) was mapped to a specific port (say 30001 for h1, 30002 for h2, etc). Then a call to any of the Nodes (using its public IP) in the cluster would proxy the port to the right host. This meant that I could call h2 from the h1 host if I used port 30002.

Reply:
lets sync up between us - there might be an important aspect we're missing when discussing this topic. I want to make sure we are on the same page and understand what we're getting from k8 here